src/vs/workbench/common/editor/filteredEditorGroupModel.ts

```diff
@@ -21,11 +20,10 @@ abstract class FilteredEditorGroupModel extends Disposable implements IReadonlyE
 		super();
 
 		this._register(this.model.onDidModelChange(e => {
-			// Only Editor Events should be filtered
-			if (e.kind !== GroupModelChangeKind.GROUP_ACTIVE && e.kind !== GroupModelChangeKind.GROUP_INDEX && e.kind !== GroupModelChangeKind.GROUP_LOCKED) {
-				const editor = e.editor ?? (e.editorIndex !== undefined ? this.model.getEditorByIndex(e.editorIndex) : undefined);
-				if (!editor || !this.contains(editor)) {
-					return;
+			const candidateOrIndex = e.editorIndex ?? e.editor;
+			if (typeof candidateOrIndex === 'number' || candidateOrIndex) {
```

Comment:
This seems weird to me... It's checking if it's not undefined, so why not just `candidateOrIndex !== undefined`, more readable

Reply:
I might be missing something, but why is this a typeof check and just `candidateOrIndex !== undefined`